app/lib/widgets/ToDoListView.dart

```diff
@@ -48,198 +52,373 @@ class _ToDoListViewState extends State<ToDoListView> {
+                                    ),
+                                  ),
+                                ],
+                              ),
+                            ),
+                          ),
+                          const Spacer(),
+                          TextButton(
+                            style: ButtonStyle(
+                              foregroundColor: MaterialStateProperty.all<Color>(
+                                ToDoTheme.secondaryTextColor,
+                              ),
+                              textStyle: MaterialStateProperty.all<TextStyle>(
+                                ToDoTheme.buttonTextStyle,
+                              ),
+                            ),
+                            onPressed: () {
+                              showDialogBox();
+                            },
+                            child: const Text('+ Add Task'),
+                          ),
+                        ],
                       ),
-                    ),
-                    onPressed: () {},
-                    child: const Text('+ Add Task'),
+                      if (todoController.expandBtn.value)
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 10),
+                          child: ListView(
+                            shrinkWrap: true,
+                            physics: const NeverScrollableScrollPhysics(),
+                            children: completedTasks,
+                          ),
+                        ),
+                    ],
                   ),
                 ),
               ],
             ),
-            if (todoController.expandBtn.value)
-              Padding(
-                padding: const EdgeInsets.symmetric(vertical: 10),
-                child: ListView(
-                  shrinkWrap: true,
-                  physics: const NeverScrollableScrollPhysics(),
-                  children: completedTasks,
-                ),
-              ),
-          ],
+          ),
         ),
       ),
     );
   }
 
   Widget buildSubtitle() {
-    if (todoController.initialExpand.value) {
-      return SizedBox(
-        child: Text(
-          widget.subtitle,
-          style: ToDoTheme.listSubtitleTextStyle.copyWith(
-            color: ToDoTheme.calendarColor,
-          ),
-        ),
-      );
-    }
     return SizedBox(
-      height: 40,
-      child: Text(
-        widget.subtitle,
-        style: ToDoTheme.listSubtitleTextStyle.copyWith(
-          color: ToDoTheme.calendarColor,
-        ),
-        maxLines: 2,
-        overflow: TextOverflow.ellipsis,
-      ),
+      child: widget.subtitle.length > 80
+          ? ExpandableText(widget.subtitle)
+          : Text(
+              widget.subtitle,
+              style: ToDoTheme.listSubtitleTextStyle.copyWith(
+                color: ToDoTheme.calendarColor,
+              ),
+            ),
+    );
+  }
+
+  showDialogBox(){
```

Comment:
Long UI method. Consider defining widget for it instead of method as they are expensive.